src/ProposalPage.js

```diff
@@ -67,47 +68,58 @@ const Content = styled(Box).attrs({
   } */
 `;
 
-export default class ProposalPage extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      showVoteControls: false,
-    };
+export default function ProposalPage({
+  proposal,
+  toggleFavorites,
+  favorite,
+  creditsBalance,
+  goBack,
+  web3Props,
+  changeAlert,
+  updateVotes,
+  voteStartTime,
+  voteEndTime,
+  history
+}) {
+  const [showVoteControls, setShowVoteControls] = useState(null);
+  const ref = useRef(null);
+
+  const onVoteControlOuterClick = () => {
+    setShowVoteControls(false);
+  };
+
+  const handleClickOutside = (event) => {
+    if (ref.current && ref.current.contains && !ref.current.contains(event.target)) {
+      onVoteControlOuterClick();
+    }
   }
-  render() {
-    const { showVoteControls } = this.state;
-    const {
-      proposal,
-      toggleFavorites,
-      favorite,
-      creditsBalance,
-      goBack,
-      web3Props,
-      changeAlert,
-      updateVotes,
-      voteStartTime,
-      voteEndTime,
-      history
-    } = this.props;
 
-    return (
-      <Container>
-        <Content as="article">
-          <HeaderBar as="nav">
-            <Back onClick={goBack}/>
-            <Star active={favorite} onClick={()=> toggleFavorites(proposal.proposalId)}/>
-          </HeaderBar>
+  useEffect(() => {
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  });
```

Comment:
Pass `handleClickOutside` in deps array for useEffect

```
useEffect(() => {
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, [handleClickOutside]);
```

Or you will remove and add listener on each render